apollo/agent/agent.py

```diff
@@ -374,7 +380,21 @@ def _execute_client_operation(
                 dict(elapsed_time=time.time() - start_time),
             ),
         )
-        return AgentResponse(result or {}, 200, operation.trace_id)
+        response = AgentResponse(result or {}, 200, operation.trace_id)
+        size = response.calculate_result_size()
```

Comment:
It looks like pre-signed urls are disabled when `operation.response_size_limit_bytes` is 0, so maybe we should check here before calling `calculate_result_size` as it seems a little bit expensive